### PR TITLE
added context in gnmi_server.handleStreamSubscriptionRequest() for proper cleanup

### PR DESCRIPTION
**Pull request by fahadnaeemkhan**
added context in `gnmi_server.handleStreamSubscriptionRequest()` to properly handle and communicate error that has happened in cache.

see https://github.com/openconfig/gnmic/pull/306/files for details

### CODE REVIEW

File: pkg/app/gnmi_server.go [hunk 1]
```diff
@@ -518,21 +519,32 @@ func (a *App) Subscribe(stream gnmi.GNMI_SubscribeServer) error {
 			errChan <- sc.stream.Send(&gnmi.SubscribeResponse{
 				Response: &gnmi.SubscribeResponse_SyncResponse{SyncResponse: true},
 			})
-			close(errChan)
+			defer close(errChan)
```
Comment (karimra) [inline pkg/app/gnmi_server.go:522]: this defer is no necessary

File: pkg/app/gnmi_server.go [hunk 2]
```diff
@@ -682,41 +690,50 @@ func (a *App) handleStreamSubscriptionRequest(sc *streamClient) {
...
-	errChan <- sc.stream.Context().Err()
+	<-ctx.Done()
+	errChan <- ctx.Err()
 	wg.Wait()
 }
 
 func (a *App) handlePolledSubscription(sc *streamClient) {
-	a.handleONCESubscriptionRequest(sc)
 	defer close(sc.errChan)
+	a.handleONCESubscriptionRequest(sc)
```
Comment (karimra) [inline pkg/app/gnmi_server.go:731]: any reason to switch the `defer` and `handleONCESubscriptionRequest` calls sequence?

File: pkg/app/gnmi_server.go (hunk 1 above)
Comment (fahadnaeemkhan) [reply pkg/app/gnmi_server.go:522]: agreed, thats typo

File: pkg/app/gnmi_server.go (hunk 2 above)
Comment (fahadnaeemkhan) [reply pkg/app/gnmi_server.go:731]: typo again

**Review by fahadnaeemkhan — COMMENTED** [review]
Let me fix theses

**Comment by karimra** [comment]
Thanks for making the changes!

**Review by karimra — APPROVED** [review]

**Pull request merged**